### DISCUSSION TITLE
MacOS-Catalyst-App brings strange error

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#11595] [Q&A]
I created a Uno-app with the latest Uno-template and tried to run it on MacOS Catalyst today. I get an exception on startup stating the "set_Name"-method does not exist on MainPage.

![image](https://user-images.githubusercontent.com/1833242/223534311-2889281e-7551-4105-b771-42da1397b057.png)

Any idea what that could be?

This is the stacktrace:

 ```
   at dfComplexitree.MainPage.InitializeComponent() in /Users/Documents/dfComplexitree/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator/MainPage_ff6af77f8dfffcbcce69d2fd905dfd97.cs:line
58
    at dfComplexitree.MainPage..ctor() in /Users/Documents/dfComplexitree/dfComplexitree/dfComplexitree/MainPage.xaml.cs:line
53
    at dfComplexitree.Mobile.MetadataBuilder_000.CreateInstance() in /Users/timparth/Documents/dfComplexitree/dfComplexitree/dfComplexitree.Mobile/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.BindableTypeProviders.BindableTypeProvidersSourceGenerator/BindableMetadata.g.cs:line
1191
    at Microsoft.UI.Xaml.Controls.Frame.CreatePageInstance(Type
sourcePageType)
    at Microsoft.UI.Xaml.PagePool.DequeuePage(Type pageType)
    at Microsoft.UI.Xaml.Controls.Frame.CreatePageInstanceCached(Type
sourcePageType)
    at
Microsoft.UI.Xaml.Controls.Frame.EnsurePageInitialized(PageStackEntry
entry)
    at Microsoft.UI.Xaml.Controls.Frame.InnerNavigate(PageStackEntry
entry, NavigationMode mode)

```

**Comment by jeromelaban** [comment]
This type of error generally means that there's a mixup in UWP and WinUI binaries, though it is supposed to be caught at build time. If you have a way to repro this, we may be able to troubleshoot.

**Reply by TopperDEL** [reply]
Hm, I'm totally lost here. I've checked my solution with a freshly created one and added one dependency after another to see if it happens wit any of them. Had no luck - the freshly created solution does not have that error.

I also checked if there is any non-WinUI-nuget in the Mobile-project, but that is also not the case. Any hints where I may search further?

I just tried to remove things from the MainPage that had "x:Name" set. That now leads to a different yet similar error with "SetRow"-MissingMethod on the Grid. So I think this won't lead me any further. I may give you access to that private github repo if that helps.

**Reply by jeromelaban** [reply]
You'll certainly need to find which dependency causes this particular issue, otherwise you can looking at a binlog to see if there are any uno.ui (and not uno.winui) dependencies that should not be there.

**Reply by TopperDEL** [reply]
Is this what you mean?

```
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.Foundation.Logging.dll 
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.UI.Composition.dll 
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.UI.Dispatching.dll 
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.UI.dll 
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.UI.FluentTheme.dll 
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.UI.FluentTheme.v1.dll 
/reference:/Users/.nuget/packages/uno.winui/4.7.44/lib/net7.0-maccatalyst/Uno.UI.FluentTheme.v2.dll 
/reference:/Users/.nuget/packages/uno.winui.lottie/4.7.44/lib/net7.0-maccatalyst/Uno.UI.Lottie.dll 
```

But these dlls come from uno.winui itself.

**Reply by TopperDEL** [reply]
Or maybe this one:

`.nuget/packages/skiasharp.views.uno.winui/2.88.3/lib/net6.0-maccatalyst13.5/SkiaSharp.Views.Windows.dll"`

**Reply by jeromelaban** [reply]
Those look like appropriate dependencies and variants. If you have custom nuget packages other than the ones provided by Uno, you may need to look into those.

**Reply by TopperDEL** [reply]
I've tried a bit with creating a new Project and adding my library to it and could reproduce the issue. It seems the csproj-file changed a bit and I removed the maccatalyst-TargetFramework completely from the file (I did not suppose to support Mac at the beginning).
So finally, the app starts on Mac.

I had to remove the MTouchExtraArgs around MONO_GC_PARAMS, otherwise I got a "could not find Accelerator.framework" or something like that. Hard stuff. :)

Now I face other problems on Mac not related to this topic here (PointerEntered not working, InlineUIContainer not supported, OnKeyDown not firing...). Hm.

**Reply by jeromelaban** [reply]
Is your library a local project or a nuget package? Is it depending on other packages?

**Reply by TopperDEL** [reply]
It is a local project. It depends on some Nuget, yes:
![image](https://user-images.githubusercontent.com/1833242/224029700-8cad0bdb-0b41-4063-8242-3dc193e136fa.png)

**Reply by TopperDEL** [reply]
Currently I don't get any PointerMoved-event unless I click somewhere.

**Reply by TopperDEL** [reply]
Ah, Pointer could be this:
https://github.com/unoplatform/uno/discussions/6918#discussioncomment-1271685
(Missing Background on SKXamlCanvas)

**Reply by jeromelaban** [reply]
Thanks. 

If you can extract a repro for the issue, it will be helpful.

It may be that PointerMoved events are not wired in properly. The Catalyst codebase is based on iOS which does have that support enabled unless the screen is touched. @dr1rrb may have some more details and we'll open an issue as needed.

**Reply by TopperDEL** [reply]
Should/could I temporarily switch to MacOs (not Catalyst) easily to overcome this issue for the moment?
